test/trimming/libsimple.jl

```diff
@@ -0,0 +1,44 @@
+module SimpleLib
+# Test the logging of entrypoints and types in a C-callable Julia library.
+
+struct CVector{T}
+    length::Cint
+    data::Ptr{T}
+end
+
+struct CVectorPair{T}
+    from::CVector{T}
+    to::CVector{T}
+end
+
+struct MyTwoVec
+    x::Int32
+    y::Int32
+end
+
+Base.@ccallable "copyto_and_sum" function badname(fromto::CVectorPair{Float32})::Float32
+    from, to = unsafe_wrap(Array, fromto.from.data, fromto.from.length), unsafe_wrap(Array, fromto.to.data, fromto.to.length)
+    copyto!(to, from)
+    return sum(to)
+end
+
+Base.@ccallable function countsame(list::Ptr{MyTwoVec}, n::Int32)::Int32
```

Comment:
Two slightly advanced (but important) features:
  - `Ref{...}` should be supported in addition to `Ptr{...}`
  - function pointers are just `Ptr{Cvoid}` ABI-wise but for C to be able to call them, we do need to export type definitions for them

Reply:
Ref is a weird case, plain C code can't make a Ref, but embedding of julia can make one

Reply:
FWIW, ccallable never supported any of those "advanced" scenarios that cfunction supports